test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java

```diff
@@ -528,6 +538,33 @@ protected String randomFetchTestFormat() {
         return null;
     }
 
+    /**
+     * Test that dimension parameter is not updateable
+     */
+    protected void registerDimensionParameter(ParameterChecker checker) throws IOException {
```

Comment:
I think this one works well. Maybe rename to `registerDimensionChecks` or something? It's not registering the parameter so much as the checks. I'm not sure.